certbot-nginx/certbot_nginx/_internal/nginxparser.py

```diff
@@ -167,8 +167,10 @@ def _coerce(self, inbound: Any) -> Tuple[Any, Any]:
                 inbound = UnspacedList(inbound)
             return inbound, inbound.spaced
 
-    def insert(self, i: int, x: Any) -> None:
+    def insert(self, i: "SupportsIndex", x: Any) -> None:
         """Insert object before index."""
+        if not isinstance(i, int):
+            raise ValueError("Only integers are supported")
```

Comment:
nit: Instead of blowing up here at runtime, what if we imported [`operator`](https://docs.python.org/3/library/operator.html) at the top of this file did:
```suggestion
        i = operator.index(i)
```

Reply:
Smart move! I was missing the point that `SupportsIndex` is a protocol. Thanks!